// ngx-http-loader/esm2020/lib/ngx-loader-interceptor.service.mjs
import { HTTP_INTERCEPTORS } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { ReplaySubject } from 'rxjs';
import { finalize } from 'rxjs/operators';
import * as i0 from "@angular/core";
export class NgxHttpLoaderInterceptor {
    constructor() {
        this.reqStatus$ = new ReplaySubject(1);
        this.pendingReqCnt = 0;
        this.forceStop = false;
        this.filterUrls = [];
        this.filterUrlPatterns = [];
        this.filterMethods = [];
        this.filterHeaders = [];
    }
    get pendingRequestCnt() {
        return this.pendingRequestCnt;
    }
    get reqStatus() {
        return this.reqStatus$.asObservable();
    }
    set setFilterUrls(urls) {
        this.filterUrls = urls;
    }
    set setUrlPatterns(regx) {
        this.filterUrlPatterns = regx;
    }
    set setFilterMethods(methods) {
        this.filterMethods = methods;
    }
    set setFilterHeaders(headers) {
        this.filterHeaders = headers;
    }
    set setForceStopReq(bool) {
        this.forceStop = bool;
    }
    checkIsFilterUrl(url) {
        return this.filterUrls.some(x => x === url);
    }
    checkUrlPatterns(url) {
        return this.filterUrlPatterns.some(x => x.test(url));
    }
    checkHttpMethod(req) {
        return this.filterMethods.some(e => {
            return e.toLowerCase() === req.method.toLowerCase();
        });
    }
    checkReqHeader(req) {
        return this.filterHeaders.some(e => {
            return req.headers.has(e);
        });
    }
    isStopReq(r) {
        return this.forceStop
            || this.checkIsFilterUrl(r.urlWithParams)
            || this.checkUrlPatterns(r.urlWithParams)
            || this.checkHttpMethod(r)
            || this.checkReqHeader(r);
    }
    intercept(request, next) {
        const isNext = this.isStopReq(request);
        if (!isNext) {
            this.pendingReqCnt++;
            if (1 === this.pendingReqCnt) {
                this.reqStatus$.next(true);
            }
        }
        return next.handle(request).pipe(finalize(() => {
            if (!isNext) {
                this.pendingReqCnt--;
                if (0 === this.pendingReqCnt) {
                    this.reqStatus$.next(false);
                }
            }
        }));
    }
}
NgxHttpLoaderInterceptor.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.1", ngImport: i0, type: NgxHttpLoaderInterceptor, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
NgxHttpLoaderInterceptor.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.0.1", ngImport: i0, type: NgxHttpLoaderInterceptor, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.1", ngImport: i0, type: NgxHttpLoaderInterceptor, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }] });
export const RequestsInterceptorProvider = [{
        provide: HTTP_INTERCEPTORS,
        useExisting: NgxHttpLoaderInterceptor,
        multi: true
    }];
