// ngx-http-loader/esm2020/lib/ngx-http-loader.module.mjs
import { NgModule } from '@angular/core';
import { RequestsInterceptorProvider } from './ngx-loader-interceptor.service';
import { ClockLoaderComponent } from './loaders/clock-loader/clock-loader.component';
import { DefaultLoaderComponent } from './loaders/default-loader/default-loader.component';
import { DualRingLoaderComponent } from './loaders/dual-ring-loader/dual-ring-loader.component';
import { EllipsisLoaderComponent } from './loaders/ellipsis-loader/ellipsis-loader.component';
import { FacebookLoaderComponent } from './loaders/facebook-loader/facebook-loader.component';
import { GearLoaderComponent } from './loaders/gear-loader/gear-loader.component';
import { GridLoaderComponent } from './loaders/grid-loader/grid-loader.component';
import { HeartLoaderComponent } from './loaders/heart-loader/heart-loader.component';
import { HourglassLoaderComponent } from './loaders/hourglass-loader/hourglass-loader.component';
import { JellyboxLoaderComponent } from './loaders/jellybox-loader/jellybox-loader.component';
import { MulticircleLoaderComponent } from './loaders/multicircle-loader/multicircle-loader.component';
import { RingLoaderComponent } from './loaders/ring-loader/ring-loader.component';
import { RippleLoaderComponent } from './loaders/ripple-loader/ripple-loader.component';
import { RollerLoaderComponent } from './loaders/roller-loader/roller-loader.component';
import { SpinnerLoaderComponent } from './loaders/spinner-loader/spinner-loader.component';
import { ThreeBounceLoaderComponent } from './loaders/three-bounce-loader/three-bounce-loader.component';
import { WaterWaveLoaderComponent } from './loaders/water-wave-loader/water-wave-loader.component';
import { NgxHttpLoaderComponent } from './ngx-http-loader.component';
import { CommonModule } from '@angular/common';
import { RotatingPlaneComponent } from './loaders/rotating-plane/rotating-plane.component';
import * as i0 from "@angular/core";
export class NgxHttpLoaderModule {
    static forRoot() {
        return {
            ngModule: NgxHttpLoaderModule,
            providers: [
                RequestsInterceptorProvider,
            ]
        };
    }
}
NgxHttpLoaderModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.1", ngImport: i0, type: NgxHttpLoaderModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgxHttpLoaderModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.1", ngImport: i0, type: NgxHttpLoaderModule, declarations: [NgxHttpLoaderComponent,
        GearLoaderComponent,
        GridLoaderComponent,
        DualRingLoaderComponent,
        ClockLoaderComponent,
        DefaultLoaderComponent,
        EllipsisLoaderComponent,
        FacebookLoaderComponent,
        HeartLoaderComponent,
        HourglassLoaderComponent,
        RingLoaderComponent,
        RippleLoaderComponent,
        RollerLoaderComponent,
        SpinnerLoaderComponent,
        WaterWaveLoaderComponent,
        ThreeBounceLoaderComponent,
        MulticircleLoaderComponent,
        JellyboxLoaderComponent,
        RotatingPlaneComponent], imports: [CommonModule], exports: [NgxHttpLoaderComponent] });
NgxHttpLoaderModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.1", ngImport: i0, type: NgxHttpLoaderModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.1", ngImport: i0, type: NgxHttpLoaderModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        NgxHttpLoaderComponent,
                        GearLoaderComponent,
                        GridLoaderComponent,
                        DualRingLoaderComponent,
                        ClockLoaderComponent,
                        DefaultLoaderComponent,
                        EllipsisLoaderComponent,
                        FacebookLoaderComponent,
                        HeartLoaderComponent,
                        HourglassLoaderComponent,
                        RingLoaderComponent,
                        RippleLoaderComponent,
                        RollerLoaderComponent,
                        SpinnerLoaderComponent,
                        WaterWaveLoaderComponent,
                        ThreeBounceLoaderComponent,
                        MulticircleLoaderComponent,
                        JellyboxLoaderComponent,
                        RotatingPlaneComponent
                    ],
                    imports: [
                        CommonModule
                    ],
                    exports: [NgxHttpLoaderComponent]
                }]
        }] });
