// ngx-http-loader/esm2020/lib/ngx-http-loader.service.mjs
import { Injectable } from '@angular/core';
import { ReplaySubject } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "./ngx-loader-interceptor.service";
export class NgxHttpLoaderService {
    constructor(reqService) {
        this.reqService = reqService;
        this.showLoader = new ReplaySubject(1);
    }
    get isShowLoader() {
        return this.showLoader.asObservable();
    }
    show() {
        this.reqService.setForceStopReq = true;
        this.showLoader.next(true);
    }
    hide() {
        this.showLoader.next(false);
        this.reqService.setForceStopReq = false;
    }
}
NgxHttpLoaderService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.1", ngImport: i0, type: NgxHttpLoaderService, deps: [{ token: i1.NgxHttpLoaderInterceptor }], target: i0.ɵɵFactoryTarget.Injectable });
NgxHttpLoaderService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.0.1", ngImport: i0, type: NgxHttpLoaderService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.1", ngImport: i0, type: NgxHttpLoaderService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.NgxHttpLoaderInterceptor }]; } });
